src/web/components/TeamMember/TeamMember.tsx

```diff
@@ -3,18 +3,77 @@ import clsx from 'clsx';
 import log from 'loglevel';
 import { useCallback, useState } from 'react';
 
-import { UserResponse } from '../../services/userAccount';
+import { UpdateTeamMemberForm, UserResponse } from '../../services/userAccount';
+import { Dialog } from '../Core/Dialog';
+import { InlineError } from '../Core/InlineError';
+import TeamMemberDialog from './TeamMemberDialog';
 
-type TeamMemberProps = { person: UserResponse; resendInvite: (id: number) => Promise<void> };
+type DeleteConfirmationDialogProps = {
+  person: UserResponse;
+  onRemoveTeamMember: () => Promise<void>;
+};
+
+function DeleteConfirmationDialog({ person, onRemoveTeamMember }: DeleteConfirmationDialogProps) {
+  const [openConfirmation, setOpenConfirmation] = useState(false);
+
+  const handleRemove = () => {
+    setOpenConfirmation(false);
+    onRemoveTeamMember();
```

Comment:
I think you need to await this - otherwise the state refresh at the top level could possibly happen before the API call to delete finishes.